### PR TITLE
Bug 1844039 - New insecure website permission prompt UI test

### PR DESCRIPTION
**Pull request by AndiAJ**
Bug 1844039 - New insecure website permission prompt UI test

Summary:
The prompt shouldn't normally be displayed on websites using `http` protocol
Even though localhost uses `http` protocol, the permission prompt is still displayed (not sure why)
Using the permission website from testapp wasn't feasible because github enforces `https` when publishing the pages
Found this website which supports both `http` and `https` protocol on the [Mozilla MDN page](https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API/Using_the_Permissions_API) and seems to be working properly.

The UI test successfully passed 100x on Firebase ✅ 

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1844039

### CODE REVIEW

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SitePermissionsRobot.kt [hunk 1]
```diff
@@ -74,14 +74,21 @@ class SitePermissionsRobot {
         assertTrue(allowPagePermissionButton.text.equals("Allow"))
     }
 
-    fun verifyLocationPermissionPrompt(url: String) {
+    fun verifyLocationPermissionPrompt(url: String, exists: Boolean) {
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SitePermissionsRobot.kt:77]: Nit: I would call the second param "shouldBeDisplayed" or "shouldExist".

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SitePermissionsTest.kt [hunk 2]
```diff
@@ -327,4 +328,26 @@ class SitePermissionsTest {
             }
         }
     }
+
+    @Test
+    fun verifyPermissionPromptForInsecureWebsiteTest() {
+        val insecureWebsite = "http://chrisdavidmills.github.io/location-finder-permissions-api"
+        val secureWebsite = "https://chrisdavidmills.github.io/location-finder-permissions-api"
```
Comment (sv-ohorvath) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SitePermissionsTest.kt:335]: I would investigate more on how to not enforce HTTPS on our github testapp and host a page there. @AaronMT is there a way to disable enforcing HTTPS on https://github.com/mozilla-mobile/testapp ?

Comment (AaronMT) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SitePermissionsTest.kt:335]: Only if we had a custom domain.

Screenshot 2023-07-18 at 10 53 00 AM

I'll look for an example of what desktop uses.

**Comment by gabrielluong** [comment]
We have moved our firefox-android repo over to [mozilla-central](https://hg.mozilla.org/mozilla-central). Please feel free to reopen a new PR against the new repo. You will have to clone the new repository and reapply your changes. Thank you again, and sorry for the inconvenience.

For more information, please see https://github.com/mozilla-mobile/firefox-android/wiki#upcoming-migration-to-mozilla-central.

**Pull request closed without merging**